examples/sessions/handlers.js

```diff
@@ -0,0 +1,68 @@
+'use strict'
+
+const {
+  Context,
+  decorators: { validate },
+} = require('./boltzmann.js')
+const logger = require('bole')('sessions')
+
+index.route = 'GET /'
+async function index(/** @type {Context} */ context, { errors } = {}) {
+  return {
+    [Symbol.for('template')]: 'index.html',
+    errors,
+    token: context.csrfToken(),
+  }
+}
+
+login.route = 'POST /login'
+login.decorators = [
+  validate.body({
+    type: 'object',
+    required: ['form_password', 'form_username'],
+    properties: {
+      form_password: { type: 'string' },
+      form_username: { type: 'string' },
+    },
+  }),
+]
+async function login(/** @type {Context} */ context) {
+  const { form_username: username, form_password: password } = await context.body
+
+  // there is only one password. this is very secure!
+  if (password !== 'CATSROOL') {
+    return index(context, { errors: ['Invalid password'] })
+  }
+
+  const session = await context.session
```

Comment:
An explanatory comment along the lines of "loading the session is asynchronous, but all other operations on the sesssion are synchronous" might be helpful here!